src/parser.ts

```diff
@@ -7984,7 +7984,7 @@ export function parseClassExpression(
 
   nextToken(parser, context);
 
-  if (((parser.token & 0x10ff) ^ 0x54) > 0x1000) {
+  if (((parser.token & 0x10ff) ^ 0x57) > Token.Keyword) {
```

Comment:
This code should be refactored using Token constants from hard coded binary.
I don't fully understand what's doing right now, but since we shift 3 numbers for many tokens, this line doesn't work right now. And my attempt of fix (by adding 3 to 0x54) didn't work. I assume the 0x54 (84) means lower part of ExtendsKeyword (now is 87).

We need some comments to explain the code, plus a proper fix. @KFlash

Reply:
I think I understand why it fails now.
The previous value for ExtendsKeyword 84 is an even number, so the lowest bit is 0, the Token.Identifier is 1, `1 ^ 0` will work.
But now ExtendsKeyword 87 has 1 at lowest bit. It doesn't work with Token.Identifier `1 ^ 1 === 0`.
That also tells me the previous code is too tricky.

Reply:
Is the logic " if this token is a keyword and not the `extends` "?

Reply:
OK, fixed. There are two places for the same code, one in parseClassDeclaration, one in parseClassExpression.

Reply:
Your way of thinking is correct. How did you fix it?

Reply:
> This code should be refactored using Token constants from hard coded binary.

How would you do that? Use of constants?

Reply:
Pls review the latest code.